src/test/java/com/thebois/models/inventory/InventoryTests.java

```diff
@@ -186,4 +190,85 @@ public void addMultipleItemsToInventory() {
         assertThat(inventory.hasItem(ItemType.LOG, 2)).isTrue();
     }
 
+    private static Stream<Arguments> calculatedDifferenceEqualToExpectedSource() {
+        return Stream.of(Arguments.of(
+            List.of(new Rock(), new Rock(), new Rock(), new Log(), new Log(), new Log()),
+            List.of(new Rock()),
+            new ArrayList<>(Arrays.asList(ItemType.LOG,
+                                          ItemType.LOG,
+                                          ItemType.LOG,
+                                          ItemType.ROCK,
+                                          ItemType.ROCK))),
+        Arguments.of(
+            List.of(new Rock(), new Rock()),
+            List.of(new Rock()),
+            new ArrayList<>(List.of(ItemType.ROCK)))
+        );
+    }
+
+    @ParameterizedTest
+    @MethodSource("calculatedDifferenceEqualToExpectedSource")
+    public void calculatedDifferenceEqualToExpected(
+        final Collection<IItem> listA,
+        final Collection<IItem> listB,
+        final ArrayList<ItemType> expectedDifference) {
+        // Arrange
+
+        final Stack<IItem> itemsA = new Stack<>();
+        final Stack<IItem> itemsB = new Stack<>();
```

Comment:
question

Why stacks?

Reply:
I misread the method addMultiple in IInventory as using stack:  `addMultiple(final List<IItem> stack)`